include/rocksdb/merge_operator.h

```diff
@@ -222,26 +222,38 @@ class MergeOperator {
   }
 };
 
-// The simpler, associative merge operator.
+// A merge operator that can be implemented as an associative binary
+// operator on values/operands. Implementations must satisfy
+//      merge_two_values(merge_two_values(a, b), c)
+//   == merge_two_values(a, merge_two_values(b, c))
+// as partial merges might substitute the latter for the former.
+//
+// A consequence of this assumption is that values must be compatible as
+// operands in associative merge operators. For example, a silly general merge
+// operator could be "total length in bytes of all operands, as a fixed
+// uint64." Although the order of operands does not matter, this would not
+// work as an associative merge operator because the operator does not know
+// whether to treat an operand as an integer length (value to add to) or an
+// arbitrary byte sequence (operand to take the length of).
 class AssociativeMergeOperator : public MergeOperator {
  public:
   ~AssociativeMergeOperator() override {}
 
-  // Gives the client a way to express the read -> modify -> write semantics
+  // Merges two values to express the read -> modify -> write semantics
   // key:           (IN) The key that's associated with this merge operation.
-  // existing_value:(IN) null indicates the key does not exist before this op
-  // value:         (IN) the value to update/merge the existing_value with
+  // left_operand:  (IN) null indicates the key does not exist before this op
+  // right_operand: (IN) the value to update/merge the left operand with
   // new_value:    (OUT) Client is responsible for filling the merge result
```

Comment:
```
// Merges two inputs to express the read -> modify -> write semantics
// key:           (IN) The key that's associated with this merge operation.
// left_input:  (IN) null indicates the key does not exist before this operation, otherwise an existing value or operand
// right_operand: (IN) the operand to merge the left_input with
// output:    (OUT) Client is responsible for filling the merge result
// here. The string that output is pointing to will be empty initially.
```

Reply:
Or you could just use `left` and `right`, as opposed to `left_input` and `right_operand`.